src/components/UI/dropdown/index.js

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState, useRef } from 'react';
+import Link from 'next/link';
+import styles from './dropdown.module.scss';
+
+const Dropdown = ({
+  isLoggedIn,
+  USER_PROFILE_URL,
+  SIGN_OUT_URL,
+  userData,
+  DEFAULT_AVATAR,
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const toggleDropdown = () => setIsOpen(!isOpen);
+  const modalClose = useRef();
+
+  const signOut = () => {
+    fetch(SIGN_OUT_URL, {
+      method: 'GET',
+      credentials: 'include',
+    }).then(() => {
+      window.location.reload();
+    });
+  };
+
+  useEffect(() => {
+    const handler = (event) => {
+      if (!modalClose.current.contains(event.target)) {
+        setIsOpen(false);
+      }
+    };
+    document.addEventListener('mousedown', handler);
+    return () => {
+      document.removeEventListener('mousedown', handler);
+    };
+  });
+
+  return (
+    <div className={styles.userGreet} ref={modalClose}>
+      <div
+        role="button"
+        tabIndex={0}
+        className={styles.dropdownHeader}
+        onClick={toggleDropdown}
+        onKeyDown={toggleDropdown}
+      >
+        <div className={styles.userGreetMsg}>
+          {isLoggedIn ? `Hello, ${userData.firstName}!` : `Hello, User!`}
+        </div>
+        <img
+          className={styles.userProfilePic}
+          src={isLoggedIn ? `${userData.profilePicture}` : `${DEFAULT_AVATAR}`}
+          alt="Profile Pic"
+        />
+      </div>
+
+      <div
+        className={` ${styles.dropDown} ${
+          isOpen ? styles.dropDownOpen : styles.dropDownClose
+        }`}
+      >
+        <div className={styles.dropdownContent}>
+          <div className={styles.dropDownContentWrapper}>
+            <div className={styles.myProfileWrapper}>
+              <Link href={USER_PROFILE_URL}>
+                <a className={styles.myProfile}>My profile</a>
+              </Link>
+            </div>
+            <div className={styles.signOutWrapper}>
+              <button
+                type="submit"
+                className={styles.signOut}
+                onClick={signOut}
+              >
+                Sign Out
+              </button>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export default Dropdown;
```

Comment:
Cant, we look to create this as a generic re-usable dropdown component? so it can be used across members? Right now it's only used for Signing users out.

Reply:
Also not sure why is this named a `dropdown` component.
I cannot see any select attribute 🤔

Reply:
I have created a re-usable dropdown component please let me know if I made any wrong choice.
I don't name this component based on any attribute I basically named it because how it will appear in the UI as you can see it the above video, please let me know if I can change the component  name from dropdown  to any better name.